### PR TITLE
Adapts WENO5 struct for GPU

### PR DESCRIPTION
**Pull request by aelligp**
Some changes in this PR are superseded by #250

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -20,6 +18,7 @@ MuladdMacro = "46d2c3a1-f734-5fdb-9937-b9b9aeba4221"
 ParallelStencil = "94395366-693c-11ea-3b26-d9b7aac5d958"
 Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
 Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
+SpecialFunctions = "276daf66-3868-5448-9aa4-cd146d93841b"
```
Comment (albert-de-montserrat) [inline Project.toml:21]:
```suggestion
```
Remove it

File: miniapps/benchmarks/thermal_diffusion/diffusion/diffusion2D_multiphase.jl [hunk 2]
```diff
@@ -73,7 +75,7 @@ function diffusion_2D(; nx=32, ny=32, lx=100e3, ly=100e3, Cp0=1.2e3, K0=3.0)
     dt       = 50 * kyr # physical time step
 
     init_mpi = JustRelax.MPI.Initialized() ? false : true
-    igg    = IGG(init_global_grid(nx, ny, 1; init_MPI = init_mpi)...)
+    # igg    = IGG(init_global_grid(nx, ny, 1; init_MPI = init_mpi)...)
```
Comment (albert-de-montserrat) [inline miniapps/benchmarks/thermal_diffusion/diffusion/diffusion2D_multiphase.jl:78]:
```suggestion
```

File: miniapps/benchmarks/thermal_diffusion/diffusion/diffusion2D_multiphase.jl [hunk 3]
```diff
@@ -129,7 +131,7 @@ function diffusion_2D(; nx=32, ny=32, lx=100e3, ly=100e3, Cp0=1.2e3, K0=3.0)
     update_phase_ratios!(phase_ratios, particles, xci, xvi, pPhases)
     # ----------------------------------------------------
 
-    @parallel (@idx ni) compute_temperature_source_terms!(thermal.H, rheology, phase_ratios.center, args)
+    @parallel (@idx ni) compute_temperature_source_terms!(thermal.H, rheology, phase_ratios, args)
```
Comment (albert-de-montserrat) [inline miniapps/benchmarks/thermal_diffusion/diffusion/diffusion2D_multiphase.jl:134]: Is it not `phase_ratios.center`?

**Comment by aelligp** [comment]
`compute_melt_fraction()` has been fixed in GP (#222)[https://github.com/JuliaGeodynamics/GeoParams.jl/pull/222] and CI  should pass as soon as GP has a new release

**Review by albert-de-montserrat — APPROVED** [review]

**Pull request merged**